sqlmesh/core/config/connection.py

```diff
@@ -1052,7 +1053,15 @@ def _connection_factory(self) -> t.Callable:
                 self.keyfile_json, scopes=self.scopes
             )
 
-        return Connector(credentials=creds).connect
+        kwargs = {
+            "credentials": creds,
+            "ip_type": self.ip_type,
+        }
+
+        if self.timeout:
+            kwargs["timeout"] = self.timeout
```

Comment:
Should we remove it from `_connection_kwargs_keys` then?

Reply:
My assumption was that it applies then to both. So we are passing it into the GCP connector class and then the connector that is created for pg8000. I'm not sure how the two interact but this at least keeps them aligned.